crates/turbopath/src/absolute_system_path_buf.rs

```diff
@@ -143,6 +151,65 @@ impl AbsoluteSystemPathBuf {
         AbsoluteSystemPathBuf(self.0.join(path.as_path()))
     }
 
+    pub fn join_literal(&self, segment: &str) -> Self {
+        AbsoluteSystemPathBuf(self.0.join(Path::new(segment)))
+    }
+
+    pub fn ensure_dir(&self) -> Result<(), io::Error> {
+        if let Some(parent) = self.0.parent() {
+            fs::create_dir_all(parent)
+        } else {
+            Ok(())
+        }
+    }
+
+    pub fn remove(&self) -> Result<(), io::Error> {
+        fs::remove_file(self.0.as_path())
+    }
+
+    pub fn set_readonly(&self) -> Result<(), io::Error> {
+        let mut perms = self.metadata()?.permissions();
+        perms.set_readonly(true);
+        fs::set_permissions(self.0.as_path(), perms)
+    }
+
+    pub fn is_readonly(&self) -> Result<bool, io::Error> {
+        Ok(self.metadata()?.permissions().readonly())
+    }
+
+    pub fn symlink_to_file<P: AsRef<Path>>(&self, to: P) -> Result<(), PathError> {
+        let system_path = to.as_ref();
+        let system_path = system_path.into_system()?;
+        #[cfg(windows)]
+        return Ok(std::os::windows::fs::symlink_file(
+            &system_path,
+            self.0.as_path(),
+        )?);
+
+        #[cfg(not(windows))]
+        return Ok(std::os::unix::fs::symlink(&system_path, self.0.as_path())?);
+    }
+
+    pub fn symlink_to_dir<P: AsRef<Path>>(&self, to: P) -> Result<(), PathError> {
+        let system_path = to.as_ref();
+        let system_path = system_path.into_system()?;
+        #[cfg(windows)]
+        return Ok(std::os::windows::fs::symlink_dir(to)?);
```

Comment:
I think this is missing an argument: https://doc.rust-lang.org/std/os/windows/fs/fn.symlink_dir.html